### PR TITLE
Lock rotation for Simplified Login View

### PR DESCRIPTION
**Pull request by zamorastp**

### CODE REVIEW

**Comment by zamorastp** [comment]
Simulator Screen Shot - iPhone 13 - 2022-01-21 at 15 47 57
Simulator Screen Shot - iPhone 13 - 2022-01-21 at 15 48 09

Still work in progress ;)

**Comment by zamorastp** [comment]
WiP
Simulator Screen Shot - iPhone 13 - 2022-01-21 at 15 47 57
Simulator Screen Shot - iPhone 13 - 2022-01-21 at 15 48 09

File: Sources/AccountSDKIOSWeb/Simplified Login/View/SimplifiedLoginViewController.swift [hunk 1]
```diff
@@ -90,7 +117,7 @@ class SimplifiedLoginViewController: UIViewController {
         
         if isPhone {
             setupConstraints()
```
Comment (Daniel-Echegaray) [inline Sources/AccountSDKIOSWeb/Simplified Login/View/SimplifiedLoginViewController.swift:119]: `setupPhoneConstraints` or `setupiPhoneConstraints` maybe is a better name

File: Sources/AccountSDKIOSWeb/Simplified Login/View/SimplifiedLoginViewController.swift [hunk 2]
```diff
@@ -188,6 +215,57 @@ class SimplifiedLoginViewController: UIViewController {
...
+    override open var preferredInterfaceOrientationForPresentation: UIInterfaceOrientation{
+        get {
+            .portrait
+        }
+    }
+    
+    override open var supportedInterfaceOrientations: UIInterfaceOrientationMask{
+        get {
+            .portrait
+        }
```
Comment (Daniel-Echegaray) [inline Sources/AccountSDKIOSWeb/Simplified Login/View/SimplifiedLoginViewController.swift:267]:
```suggestion
        if isPhone{
            return .portrait
        } else {
            return .all
        }
```

Comment (Daniel-Echegaray) [reply Sources/AccountSDKIOSWeb/Simplified Login/View/SimplifiedLoginViewController.swift:267]: The os will call this function on iPad if the app target has [disabled multitasking ](https://stackoverflow.com/a/35275848) and if the viewcontroller is presented full screen (we have it as formSheet though for iPad). 

Might clarify the intent to return .all when device is ipad.

**Review by Daniel-Echegaray — APPROVED** [review]
Looks good! Nice work 🚀  

I've just added a few suggestions for which I think could make the code clearer.

File: Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift [hunk 3]
```diff
@@ -117,7 +117,8 @@ struct SimplifiedLoginUIFactory {
     
     private static func commonSetup(completion: @escaping LoginResultHandler, client: Client, assertionFetcher: SimplifiedLoginFetching,  viewModel: SimplifiedLoginViewModel) -> UIViewController {
         let s = SimplifiedLoginViewController(viewModel: viewModel )
-        let nc = UINavigationController()
+        let nc = SimplifiedLoginNavigationController()
+        nc.view.backgroundColor = .clear
```
[the remark this answers is not in the document]
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginUIFactory.swift:121]: sure 👍

File: Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginManager.swift [hunk 4]
```diff
@@ -156,6 +156,8 @@ extension SimplifiedLoginManager {
         if Self.isPad {
             simplifiedLoginViewController.modalPresentationStyle = .formSheet
             simplifiedLoginViewController.preferredContentSize = .init(width: 450, height: 424)
+        } else {
+            simplifiedLoginViewController.modalPresentationStyle = .overFullScreen
         }
```
[the remark this answers is not in the document]
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/SimplifiedLoginManager.swift:161]: Moved to SimplifiedNavigationController. Pure is embedded in nav controller and returned from factory

File: Sources/AccountSDKIOSWeb/Simplified Login/View/SimplifiedLoginViewController.swift [hunk 5]
```diff
@@ -90,7 +117,7 @@ class SimplifiedLoginViewController: UIViewController {
         
         if isPhone {
             setupConstraints()
-            setupNavigationBar()
+            //setupNavigationBar()
```
[the remark this answers is not in the document]
Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/View/SimplifiedLoginViewController.swift:120]: done

File: Sources/AccountSDKIOSWeb/Simplified Login/View/SimplifiedLoginNavigationController.swift [hunk 6]
```diff
@@ -1,9 +1,18 @@
...
     override func viewDidLoad() {
         super.viewDidLoad()
+        view.backgroundColor = .clear
+        
+        if Self.isPad {
+            modalPresentationStyle = .formSheet
+            preferredContentSize = .init(width: 450, height: 424)
+        } else {
+            modalPresentationStyle = .overFullScreen
+        }
```
Comment (Daniel-Echegaray) [inline Sources/AccountSDKIOSWeb/Simplified Login/View/SimplifiedLoginNavigationController.swift:15]: Wasn't this logic setting the SimplifiedLoginViewController before? This change, is applying it to the SimplifiedLoginNavigationController

Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Simplified Login/View/SimplifiedLoginNavigationController.swift:15]: No, it never was in SimplifiedLoginViewController. I replied to your first comment that this should go to navigation controller.

**Pull request merged**